index.go

```diff
@@ -252,3 +261,27 @@ type IndexBuilder interface {
 	Index(doc Document) error
 	Close() error
 }
+
+type SynonymTermReader interface {
```

Comment:
Let's rename this to ThesaurusTermReader to align in naming?

Reply:
ack